robotoff/scheduler/__init__.py

```diff
@@ -44,38 +44,39 @@
 
 # Note: we do not use with_db, for atomicity is handled in annotator
 def process_insights():
-    processed = 0
-    for insight in (
-        ProductInsight.select()
-        .where(
-            ProductInsight.annotation.is_null(),
-            ProductInsight.process_after.is_null(False),
-            ProductInsight.process_after <= datetime.datetime.utcnow(),
-        )
-        .iterator()
-    ):
-        try:
-            annotator = InsightAnnotatorFactory.get(insight.type)
-            logger.info(
-                "Annotating insight %s (product: %s)", insight.id, insight.barcode
+    with db.connection_context():
+        processed = 0
+        for insight in (
+            ProductInsight.select()
+            .where(
+                ProductInsight.annotation.is_null(),
+                ProductInsight.process_after.is_null(False),
+                ProductInsight.process_after <= datetime.datetime.utcnow(),
             )
-            annotation_result = annotator.annotate(insight, 1, update=True)
-            processed += 1
-
-            if annotation_result == UPDATED_ANNOTATION_RESULT and insight.data.get(
-                "notify", False
-            ):
-                slack.NotifierFactory.get_notifier().notify_automatic_processing(
-                    insight
+            .iterator()
+        ):
+            try:
+                annotator = InsightAnnotatorFactory.get(insight.type)
+                logger.info(
+                    "Annotating insight %s (product: %s)", insight.id, insight.barcode
+                )
+                annotation_result = annotator.annotate(insight, 1, update=True)
+                processed += 1
+
+                if annotation_result == UPDATED_ANNOTATION_RESULT and insight.data.get(
+                    "notify", False
+                ):
+                    slack.NotifierFactory.get_notifier().notify_automatic_processing(
+                        insight
+                    )
```

Comment:
So that even if there is one error during the annotation of one product (caused by Product Opener timeout for example), we still keep the annotation status for those that have been annotated.

Reply:
If you put it in db.atomic() you will have this more surely